src/gui/hotkeyConfig.cpp

```diff
@@ -310,3 +335,42 @@ void HotkeyConfigItem::load(string key_config)
         }
     }
 }
+
+bool HotkeyConfig::setHotKey(std::string work_cat, std::pair<string,string> key, std::string new_value)
+{
+
+    // needs test if new_value is part of the sfml_list
+    for (std::pair<string, sf::Keyboard::Key> sfml_key : sfml_key_names) {
+
+        if (hotkeys.iequals(sfml_key.first, new_value) || new_value == "") {
+            for (HotkeyConfigCategory &cat : categories) {
+                if (cat.name == work_cat) {
+                    int i = 0;
+                    for (HotkeyConfigItem &item : cat.hotkeys) {
+                        if (key.first == std::get<0>(item.value)) {
+
+                            item.load(new_value);
+                            item.value = std::make_tuple(std::get<0>(item.value), new_value);
+
+                            return true;
+                        }
+                        i++;
+                    }
+                }
+            }
+        }
+    }
+    return false;
+}
+
+bool HotkeyConfig::iequals(const string& a, const string& b)
```

Comment:
This function has no reason to be in the hotkey config.
The strings are also the class exposed from serious proton, which have a "lower" function. So the whole code can be replaced with "if (a.lower() == b.lower())"